### PR TITLE
Make _resultDependsOn/_resultDependsOnSelf available in SIL

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

File: lib/SIL/IR/SILPrinter.cpp [hunk 1]
```diff
@@ -686,6 +688,8 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
       *this << "@reborrow ";
     if (i.IsEscaping)
       *this << "@pointer_escape ";
+    if (i.HasResultDependsOn)
+      *this << "_resultDependsOn ";
```
Comment (atrick) [inline lib/SIL/IR/SILPrinter.cpp:692]: Should this be `@_resultDependsOn`?

File: lib/SIL/IR/SILPrinter.cpp [hunk 2]
```diff
@@ -3416,6 +3416,10 @@ void SILFunction::print(SILPrintContext &PrintCtx) const {
   if (needsStackProtection())
     OS << "[stack_protection] ";
 
+  if (hasResultDependsOnSelf()) {
+    OS << "_resultDependsOnSelf ";
```
Comment (atrick) [inline lib/SIL/IR/SILPrinter.cpp:3420]: should this be `[_resultDependsOnSelf]`?

**Review by atrick — APPROVED** [review]
Thanks. At the SIL level, I wonder how hard it would be to use a more expressive lifetime-dependence model rather than modeling a temporary workaround.

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test Linux Platform

**Comment by meg-gupta** [comment]
@swift-ci smoke test macOS platform

**Pull request merged**